docs: Add imports guide for contributors (#31353)
akka/akka: CONTRIBUTING.md

PR validation includes checking that the Scala sources are formatted and will fail if they are not.

Akka prefer the flatten imports rather than grouped, which helps reducing merge conflicts. If you are using IDEA, you can set 

> the option with: `Preferences` -> `Code Style` -> `Scala` -> `Imports` -> `Collect Imports with the same prefix into one import`.

Reviewer: Should that be disable/unset, because we do not want to collect all imports into one?

PR validation includes checking that the Scala sources are formatted and will fail if they are not.

Akka prefer flatten imports rather than grouped, which helps reduce merge conflicts. 
If you are using IDEA, you can disable it with: `Preferences` -> `Code Style` -> `Scala` -> `Imports` -> `Collect Imports with the same prefix into one import` 

> unselecting the checkbox.

Reviewer 2: I  updated it on how to disable it.

> Akka prefer flatten imports rather than grouped, which helps reduce merge conflicts. 
> If you are using IDEA, you can disable it with: `Preferences` -> `Code Style` -> `Scala` -> `Imports` -> `Collect Imports with the same prefix into one import` 
> unselecting the checkbox.

Reviewer 3: ```suggestion
Akka prefers flattened imports rather than grouped, which helps reduce merge conflicts. 
If you are using IntelliJ IDEA, you can disable it by unchecking: `Preferences` -> `Code Style` -> `Scala` -> `Imports` -> `Merge imports with the same prefix into one statement`.
```
